client/checkout/classic/index.js

```diff
@@ -353,6 +386,60 @@ jQuery( function ( $ ) {
 		return false;
 	};
 
+	/**
+	 * Submits checkout form via AJAX to create order and uses custom
+	 * redirect URL in AJAX response to request payment confirmation from UPE
+	 *
+	 * @param {Object} $form The jQuery object for the form.
+	 * @return {boolean} A flag for the event handler.
+	 */
+	const handleUPECheckout = async ( $form ) => {
+		if ( ! paymentIntent ) {
+			return;
+		}
+
+		if ( ! isUPEComplete ) {
+			// If UPE fields are not filled, confirm payment to trigger validation errors
+			const { error } = await api.getStripe().confirmPayment( {
+				element: upeElement,
+				confirmParams: {
+					return_url: '',
+				},
+			} );
+			showError( error.message );
+
+			return;
+		}
+
+		blockUI( $form );
+		// Create object where keys are form field names and keys are form field values
+		const formFields = $form.serializeArray().reduce( ( obj, field ) => {
+			obj[ field.name ] = field.value;
+			return obj;
+		}, {} );
+
+		try {
+			const response = await api.processCheckout(
+				paymentIntent.id,
+				formFields
+			);
+			const redirectUrl = response.redirect_url;
+			const { error } = await api.getStripe().confirmPayment( {
+				element: upeElement,
+				confirmParams: {
+					// eslint-disable-next-line camelcase
+					return_url: redirectUrl,
+				},
+			} );
+			if ( error ) {
+				throw error.message;
+			}
+		} catch ( error ) {
+			$form.removeClass( 'processing' ).unblock();
+			showError( error );
```

Comment:
something is not quite right with the error handling here. If you try to checkout but force an HTTP error it will show on the page `[object] object` instead of a friendly error message. Easiest way to reproduce is to activate xdebug and set a breakpoint but then let the request timeout. On my setup it timesout after 5 mins with a 503 error code but the page just shows this:
![image](https://user-images.githubusercontent.com/1534605/119571601-c5f2ad80-bd6e-11eb-82ca-0d5f59d1cc80.png)

Reply:
I added a more readable error over [here](https://github.com/Automattic/woocommerce-payments/blob/41a0e2a9d8d9460c942ced8ddcc3ebd710695b74/client/checkout/api/index.js#L352), but I'm willing to wobble on the specifics on the message itself if you have any better ideas.